Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp

```diff
@@ -921,6 +943,13 @@ class ObjectAllocationSinkingPhase : public Phase {
         return result;
     }
 
+    Node* ensureConstant(unsigned constant)
+    {
+        return m_constants.ensure(constant, [&]() {
+            return m_insertionSet.insertConstant(0, m_graph.block(0)->at(0)->origin, jsNumber(constant));
```

Comment:
This is not correct since `m_insertionSet` needs to be executed with root block, but it isn't. We should have different InsertionSet `m_rootInsertionSet`, and we need to run `m_rootInsertionSet.execute(rootBlock)` later.